docs/docs/development/hardware-integration/pinctrl.mdx

```diff
@@ -0,0 +1,336 @@
+
+- `bias-disable`: Disable pull-up/down (default behavior, not required).
+- `bias-pull-up`: Enable pull-up resistor.
+- `bias-pull-down`: Enable pull-down resistor.
+- `low-power-enable`: Configure pin as an input with input buffer disconnected.
+
+Note that bias options are mutually exclusive.
+
+There is an additional child node for the "sleep" state, configuring the pins for low power. More on states will be explained later.
+</TabItem>
+<TabItem value="rp2040">
+```dts title="<board>-pinctrl.dtsi"
+&pinctrl {
+    /* configuration for spi0 device, default state */
+    spi0_default: spi0_default {
+        /* group 1 */
+        group1 {
+            /* configure P18 as SPI0 clock, P19 as SPI0 MOSI */
+            pinmux = <SPI0_SCK_P18>, <SPI0_TX_P19>;
+        };
+        /* group 2 */
+        group2 {
+            /* configure P16 as SPI0 MISO */
+            pinmux = <SPI0_RX_P16>;
+            /* enable input on pin 1 */
+            input-enable;
+        };
+    };
+
+    /* configuration for an spi device using PIO0, default state  */
+    pio0_spi_default: pio0_spi_default {
+        /* group 1 */
+        group1 {
+            /* Configure P13 and P14 for PIO, to be used for MOSI and SCK */
+            pinmux = <PIO0_P13>, <PIO0_P14>;
+        };
+        /* group 2 */
+        group2 {
+            /* Configure P16 for PIO to be used for MISO with input */
+            pinmux = <PIO0_P12>;
+            input-enable;
+        };
+    };
+};
+````
+
+The values that can be listed in `pinmux` to assign functionality to pins are listed in `zephyr/include/zephyr/dt-bindings/pinctrl/rpi-pico-rp2040-pinctrl.h`.
+
+The following pin properties can be assigned to groups:
+
+- `bias-disable`: Disable pull-up/down (default, not required).
+- `bias-pull-up`: Enable pull-up resistor.
+- `bias-pull-down`: Enable pull-down resistor.
+- `input-enable`: Enable input from the pin.
+- `input-schmitt-enable`: Enable input hysteresis.
+- `drive-strength`: Set the drive strength of the pin, in milliamps. Possible values are: 2, 4, 8, 12 (default: 4mA)
+- `slew-rate`: If set to 0, slew rate is set to slow. If set to 1, it is set to fast.
+
+</TabItem>
+</Tabs>
+
+## Driver/Bus Node
+
+Once pin control for a driver/bus has been defined, you'll need to adjust another node defining the driver/bus. This adjustment can be done in a number of places by convention:
+
+- If defining a unique board, `<board>.dts`
+- If defining boards with multiple revisions/versions that share pin control, `<board>-common.dtsi` (which is then included by each `<board>_<revision>.dtsi`)
+- If [configuring boards for a shield](#boards-shields-and-modules), directly in the `<board>.overlay` file
+
+You'll want to identify the correct node for you to be changing. The nRF52840 has nodes defined in `dts/arm/nordic/nrf52840.dtsi`, while the RP2040 has nodes defined in `dts/arm/rpi_pico/rp2040.dtsi`. **Make sure the node that you're using is not already in use elsewhere.**
```

Comment:
What is the reason for the "not already in use" remark? I didn't understand it.

Reply:
Both the rp2040 and the nrf52840 (and I'd imagine most other MCUs) have a limited number of each bus that can be used, even though the pins that can be tied to each bus are flexible. The nrf52 has a maximum of two I2C buses, and xiao ble sense (Zephyr's 3.7 version, not ours, ours isn't set up for this) uses one of these internally for the gyroscope. If you were to use the same node for I2C as the gyroscope uses, then you'd need to override pinctrl and disable the gyroscope. 

While the gyroscope is irrelevant, this might be relevant for a different board or if someone is defining multiple of the same bus on one device.

Reply:
I see, I interpreted this as more like "don't modify `spi0` if it was already defined" which you will want to do in some/most cases. (There might not be any devices on the bus, or you don't care about them.) It might be better to add a generic warning like "pay attention to what bus you are modifying, which might already have devices defined for it".